edb/edgeql/tracer.py

```diff
@@ -578,8 +585,21 @@ def trace_Path(
                         # process it on the next step.
                         pass
                     else:
-                        # otherwise we cannot say anything about the target,
-                        # so bail.
+                        # No type intersection, so the only type that
+                        # it can be is "Object", which is trivial.
+                        # However, we need to make it dependent on
+                        # every link of the same name now.
+                        for fqname, ptr in ctx.objects.items():
```

Comment:
You need to check that `ptr` is actually a `Pointer` here.

Reply:
Added the `Pointer` check.